kobo/apps/audit_log/tests/test_signals.py

```diff
@@ -0,0 +1,142 @@
+import contextlib
+from unittest.mock import patch
+
+from allauth.account.models import EmailAddress
+from django.contrib.auth.signals import user_logged_in
+from django.test import override_settings
+from django.urls import resolve, reverse
+from trench.utils import get_mfa_model
+
+from kobo.apps.audit_log.models import AuditAction, AuditLog
+from kobo.apps.audit_log.signals import create_access_log
+from kobo.apps.kobo_auth.shortcuts import User
+from kpi.tests.base_test_case import BaseTestCase
+
+
+@contextlib.contextmanager
+def skip_login_access_log():
+    """
+    Context manager for skipping the creation of an access log on login
+
+    Disconnects the method that creates access logs from the user_logged_in signal within the contextmanager block.
+    Useful when you want full control over the audit logs produced in a test.
+    """
+    user_logged_in.disconnect(create_access_log)
+    yield
+    user_logged_in.connect(create_access_log)
+
+
+class AuditLogSignalsTestCase(BaseTestCase):
+    """
+    Class for testing that logins produce AuditLogs.
+
+    Here we just test that AuditLogs are produced, not necessarily what they contain. More tests for what they contain
+    are in test_models.py. Also, AuditLogs for SSO logins are tested as part of the SSO tests
+    to avoid copying lots of complicated setup.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user = User.objects.create_user(
+            'user', 'user@example.com', 'pass'
+        )
+        cls.user.backend = 'django.contrib.auth.backends.ModelBackend'
+        cls.user.save()
+
+    @patch('kobo.apps.audit_log.signals.AuditLog.create_access_log_for_request')
+    def test_audit_log_created_on_login(self, patched_create):
+        """
+        Basic plumbing test to make sure the signal is hooked up
+        """
+        self.client.login(username='user', password='pass')
+        patched_create.assert_called_once()
+
+    @override_settings(ACCOUNT_EMAIL_VERIFICATION='none')
+    def test_simple_login(self):
+        count = AuditLog.objects.count()
+        self.assertEqual(count, 0)
+        user = AuditLogSignalsTestCase.user
+        data = {
+            'login': 'user',
+            'password': 'pass',
+        }
+        with self.settings(ACCOUNT_EMAIL_VERIFICATION='none'):
```

Comment:
Maybe it's redundant because of the decorator?

Reply:
My b. Good catch.